### PR TITLE
Disable k/k way of log dumping if test-infra way is enabled

### PR DESCRIPTION
**Pull request by tosi3k**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Do not invoke k/k log dumping script if the test job already leverages the log dumping mechanism from test-infra repo.

#### Which issue(s) this PR fixes:
No issue is opened for that.

#### Special notes for your reviewer:
/assign @jkaniuk
/assign @marseel 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by tosi3k** [comment]
/sig scalability
/cc @YangLu1031 

Relevant Slack discussion: https://kubernetes.slack.com/archives/C09QZTRH7/p1621986566037600.

**Comment by pjh** [comment]
@spiffxp perhaps you could review and approve this with your sig-testing-approvers power?

Please see the Slack discussion that Antoni shared, https://kubernetes.slack.com/archives/C09QZTRH7/p1621986566037600.

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ func EnsureLoadBalancerResourcesDeleted(ip, portRange string) error {
 // CoreDump SSHs to the master and all nodes and dumps their logs into dir.
 // It shells out to cluster/log-dump/log-dump.sh to accomplish this.
 func CoreDump(dir string) {
-	if TestContext.DisableLogDump {
+	if TestContext.DisableLogDump || os.Getenv("USE_TEST_INFRA_LOG_DUMPING") == "true" {
```
Comment (spiffxp) [inline test/e2e/framework/util.go:1188]: I would rather not tangle in test-infra concerns.  Can your script/job not add `--disable-log-dump`?

**Review by spiffxp — CHANGES_REQUESTED** [review]

**Comment by pjh** [comment]
/close

We submitted https://github.com/kubernetes-sigs/windows-testing/pull/271 so that this is not needed for our test jobs (Windows-on-GCE).

I don't know if other test jobs will need similar changes. If something is being deprecated, we should provide accurate and complete instructions for migrating to the new thing. @tosi3k please address https://github.com/kubernetes/kubernetes/issues/96915#issuecomment-866428399, thanks.

**Pull request closed without merging**